src/brokers/metrics/components/CardBrokerCPUUsageMetrics/CardBrokerCPUUsageMetrics.container.tsx

```diff
@@ -0,0 +1,74 @@
+import { FC, useEffect, useState } from 'react';
+import { CardBrokerCPUUsageMetrics } from './CardBrokerCPUUsageMetrics';
+import {
+  usePrometheusPoll,
+  PrometheusEndpoint,
+  PrometheusResponse,
+} from '@openshift-console/dynamic-plugin-sdk';
+import { parsePrometheusDuration } from '../../../../utils';
+import { getMaxSamplesForSpan } from '../../utils';
+
+export type CardBrokerCPUUsageMetricsContainerProps = {
+  name: string;
+  namespace: string;
+  defaultSamples?: number;
+  timespan?: number;
+  fixedEndTime?: number;
+};
+
+type AxisDomain = [number, number];
+
+export const CardBrokerCPUUsageMetricsContainer: FC<
+  CardBrokerCPUUsageMetricsContainerProps
+> = ({ name, namespace = 'deafult', defaultSamples = 300, timespan }) => {
+  //states
+  const [xDomain] = useState<AxisDomain>();
+  // For the default time span, use the first of the suggested span options that is at least as long
+  // as defaultTimespan
+  const defaultSpanText = '30m';
+  const [span, setSpan] = useState(parsePrometheusDuration(defaultSpanText));
+  // If we have both `timespan` and `defaultTimespan`, `timespan` takes precedence
+  // Limit the number of samples so that the step size doesn't fall below minStep
+  const maxSamplesForSpan = defaultSamples || getMaxSamplesForSpan(span);
+  const [samples, setSamples] = useState(maxSamplesForSpan);
+
+  // Define this once for all queries so that they have exactly the same time range and X values
+  const now = Date.now();
+  const endTime = xDomain?.[1];
+
+  // If provided, `timespan` overrides any existing span setting
+  useEffect(() => {
+    if (timespan) {
+      setSpan(timespan);
+      setSamples(defaultSamples || getMaxSamplesForSpan(timespan));
+    }
+  }, [defaultSamples, timespan]);
+
+  const [result, loaded] = usePrometheusPoll({
+    endpoint: PrometheusEndpoint.QUERY_RANGE,
+    query: `pod:container_cpu_usage:sum{pod='${
```

Comment:
A comment for later - would it be possible to consolidate our metrics strings somewhere as a single go-to source of 'truth' and then refer to them as required here? Does that give us the opportunity to be more flexible and to have more reusable code around this location?